File: app/src/main/java/com/mnj/java_solutions/NthFibonacciNumber13082023.java
package com.mnj.java_solutions;

/**
 * Given a positive integer n, find the nth fibonacci number.
 * Since the answer can be very large, return the answer modulo 1000000007
 * <p>
 * Input:
 * n = 5
 * Output:
 * 5
 * Explanation:
 * 5 is the 5th number of fibonacci series..
 **/
public class NthFibonacciNumber13082023 {

    public void nthFibonacci() {
        int n = 5;
        int m = 1000000007;

        int a = 0;
        int b = 1;
        int ans = a + b;

        for (int i = 2; i <= n; i++) {
            ans = (a + b);
            a = b;
            b = ans;
        }
        System.out.println("==>> nth fibonacci number is  :" + ans);
    }
}

File: app/src/main/java/com/mnj/java_solutions/NthFibonacciNumber13082023.kt
package com.mnj.java_solutions

/**
 * Given a positive integer n, find the nth fibonacci number.
 * Since the answer can be very large, return the answer modulo 1000000007
 *
 *
 * Input:
 * n = 5
 * Output:
 * 5
 * Explanation:
 * 5 is the 5th number of fibonacci series..
 */
class NthFibonacciNumber13082023 {
    fun nthFibonacci() {
        val n = 5
        val m = 1000000007
        var a = 0
        var b = 1
        var ans = a + b
        for (i in 2..n) {
            ans = a + b
            a = b
            b = ans
        }
        println("==>> nth fibonacci number is  :$ans")
    }
}
